execution/shuffle_strategy.go

```diff
@@ -0,0 +1,99 @@
+package execution
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/twmb/murmur3"
+
+	"github.com/cube2222/octosql"
+)
+
+type ShuffleStrategyPrototype interface {
+	Get(ctx context.Context, variables octosql.Variables) (ShuffleStrategy, error)
+}
+
+type KeyHashingStrategyPrototype struct {
+	keyExpression []Expression
```

Comment:
is there any reason that in physical there is `Key` and here is `keyExpression`?

Reply:
I'll fix it.